pkg/controller/appinst_api.go

```diff
@@ -1236,12 +1239,39 @@ func (s *AppInstApi) createAppInstInternal(cctx *CallContext, in *edgeproto.AppI
 	reqCtx, reqCancel := context.WithTimeout(ctx, s.all.settingsApi.Get().CreateAppInstTimeout.TimeDuration())
 	defer reqCancel()
 
-	err = edgeproto.WaitForAppInstInfo(reqCtx, &in.Key, s.store,
-		edgeproto.TrackedState_READY,
-		CreateAppInstTransitions, edgeproto.TrackedState_CREATE_ERROR,
-		"Created AppInst successfully", cb.Send,
-		edgeproto.WithCrmMsgCh(sendObj.crmMsgCh),
-	)
+	crmAction := func() error {
```

Comment:
I am not sure why we need to define `crmAction`, since we just call it right away - this can just be inlined here.

Reply:
It's to simplify the error handling. You see there's a lot of error paths there, and we need to be able to ignore any of them if IGNORE_CRM_ERRORS is set.

Reply:
Ok, I see.